devices/ICub/src/ICub.cpp

```diff
@@ -0,0 +1,388 @@
+        if (icubImpl->leftHandFTPort.getFirstDataFlag() || icubImpl->rightHandFTPort.getFirstDataFlag()) {
+            auto nonConstThis = const_cast<ICubForceTorque6DSensor*>(this);
+            nonConstThis->setStatus(wearable::sensor::SensorStatus::Ok);
+        }
+
+    }
+
+    ~ICubForceTorque6DSensor() override = default;
+
+     void setStatus(const wearable::sensor::SensorStatus status) { m_status = status; }
+
+    // ==============================
+    // IForceTorque6DSensor interface
+    // ==============================
+    bool getForceTorque6D(Vector3& force3D, Vector3& torque3D) const override {
+        if (!icubImpl) {
+            return false;
+        }
+
+        std::vector<double> wrench;
+
+        // Reading wrench from WBD ports
+        if (this->m_name == icubImpl->wearableName + sensor::IForceTorque6DSensor::getPrefix() + "leftWBDFTSensor") {
+            icubImpl->leftHandFTPort.useCallback();
+            wrench = icubImpl->leftHandFTPort.getWrench();
+
+            icubImpl->timeStamp.time = yarp::os::Time::now();
+        }
+        else if(this->m_name == icubImpl->wearableName + sensor::IForceTorque6DSensor::getPrefix() + "rightWBDFTSensor") {
+            icubImpl->rightHandFTPort.useCallback();
+            wrench = icubImpl->rightHandFTPort.getWrench();
+
+            icubImpl->timeStamp.time = yarp::os::Time::now();
+        }
+
+
+        // Check the size of wrench values
+        if(wrench.size() == 6) {
+            force3D[0] = wrench.at(0);
+            force3D[1] = wrench.at(1);
+            force3D[2] = wrench.at(2);
+
+            torque3D[0] = wrench.at(3);
+            torque3D[1] = wrench.at(4);
+            torque3D[2] = wrench.at(5);
+        }
+        else {
+            yWarning() << LogPrefix << "Size of wrench read is wrong. Setting zero values instead.";
+            force3D.fill(0.0);
+            torque3D.fill(0.0);
+
+            // Set sensor status to error if wrong data is read
+            //auto nonConstThis = const_cast<ICubForceTorque6DSensor*>(this);
+            //nonConstThis->setStatus(wearable::sensor::SensorStatus::Error);
+        }
+
+        return true;
+    }
+};
+
+// ==========================================
+// ICub constructor/destructor implementation
+// ==========================================
+ICub::ICub()
+    : pImpl{new ICubImpl()}
+{}
+
+ICub::~ICub() = default;
+
+// ======================
+// DeviceDriver interface
+// ======================
+bool ICub::open(yarp::os::Searchable& config)
+{
+    yInfo() << LogPrefix << "Starting to configure";
+
+    // Configure clock
+    if (!yarp::os::Time::isSystemClock())
+    {
+        yarp::os::Time::useSystemClock();
+    }
+
+    yarp::os::Bottle wbdHandsFTSensorsSet = config.findGroup("wbd-hand-ft-sensors");
+    if(wbdHandsFTSensorsSet.isNull()) {
+        yError() << LogPrefix << "REQUIRED parameter <wbd-hand-ft-sensors> NOT found";
+        return false;
+    }
+    else {
+
```

Comment:
Configure `clang-format` on your IDE before merging and format all the sources of this PR. For instance, this space should be automatically removed.

Reply:
Do you have any `clang-format` template used for the `wearables` project?? @diegoferigo

Reply:
It is in the root of the project ([.clang-format](https://github.com/robotology-playground/wearables/blob/master/.clang-format)). Modern IDE with clang-format support should use it out-of-the-box.

Reply:
Done